modules/Database/Arangojs.js

```diff
@@ -863,6 +863,27 @@ class ArangoJS {
         return count;
     }
 
+    /**
+     * This method will leave only encryption for selected color, datasetId and offerId
+     * Rest of the encryption data will be removed
+     * @param datasetId
+     * @param offerId
+     * @param leaveColor
+     * @returns {Promise<void>}
+     */
+    async removeUnnecessaryEncryptionData(datasetId, offerId, leaveColor) {
+        const queryString = `LET datasetMetadata = DOCUMENT('ot_datasets', @datasetId)
+for v in DOCUMENT('ot_vertices', datasetMetadata.vertices)
```

Comment:
There could be encrypted data also in the edges, 
When the dataset is encrypted, relation properties are also encrypted (modules/ImportUtilities.js:326)
Then those encrypted relation properties are stored in the edge, not the vertex of the graph (modules/worker/graph-converter-worker.js:222)